feat: adding default error view on iOS (#356)
ZupIT/docs-beagle: content/en/resources/customization/beagle-for-ios/custom-beagle-navigation-controller.md

The method `serverDrivenStateDidChange()` is the entry point to handle screen state changes.

> The default implementation shows an `UIActivityIndicatorView` when the screen is loading and also shows a default error screen with errors treatment and a retry button.

Reviewer: ```suggestion
The default implementation shows a `UIActivityIndicatorView` when the screen is loading. It also shows a default error screen with errors treatment and a retry button.
```

The method `serverDrivenStateDidChange()` is the entry point to handle screen state changes.

> The default implementation shows an "ActivityIndicator" when the screen is loading and shows an default error view that handles error and retry.

Reviewer 2: ```suggestion
The default implementation shows an "ActivityIndicator" when the screen is loading and also shows a default error screen with errors treatment and a retry button. 
```

> When overriding, if you want to preserve loading behavior, a "super" implementation should be called or you can customize a loading and handle error behavior by yourself.

Reviewer 2: ```suggestion
When overriding, if you want to preserve a loading behavior, a "super" implementation should be called or you can customize a loading and handle the error behavior by yourself.
```

## Example

> ### **Step 1: Create a class inherited from BeagleNavigationController**
> 
> You will see below a complete example of a class that inherits from `BeagleNavigationController`. In it, the `serverDrivenStateDidChange` method is overriden and it handles error state differently.

Reviewer 2: ```suggestion
### **Step 1: Create a class inherited from BeagleNavigationController**

You will see below an example of a class inherited from `BeagleNavigationController`. The `serverDrivenStateDidChange` method is overridden in it and handles the error state differently.
```

You can register as many **BeagleNavigationController** as you need.
{{% /alert %}}

### **Step 3: Use BeagleScreenViewController with controllerId**

> Now when you instantiating the BeagleScreenViewController, you can pass the `controllerId` "CustomBeagleNavigation" so that your BeagleScreenViewController use your `CustomBeagleNavigationController`.

Reviewer: ```suggestion
Now, when you are instantiating the BeagleScreenViewController, you can pass the `controllerId` "CustomBeagleNavigation" so that your BeagleScreenViewController uses your `CustomBeagleNavigationController`.
```
